src/gui/profilepagewidget.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "ocsprofileconnector.h"
+
+#include <QBoxLayout>
+#include <QLabel>
+
+#include <cstddef>
+
+namespace OCC {
+
+class ProfilePageWidget : public QWidget
+{
+    Q_OBJECT
+public:
+    explicit ProfilePageWidget(QWidget *parent = nullptr);
+
+    void setProfileConnector(std::unique_ptr<OcsProfileConnector> profileConnector, const QString &userId);
+
+private:
+    void onHovercardFetched();
+    void onIconLoaded(const std::size_t &hovercardActionIndex);
+
+    void recreateLayout();
+    void resetLayout();
+    void createLayout();
+
+    std::unique_ptr<OcsProfileConnector> _profileConnector;
+
+    QVBoxLayout *_mainLayout{};
+    std::vector<QLabel *> _profilePageButtonIcons;
```

Comment:
@FlexW I am a bit surprised why `QVector` wouldn't suffice.

Reply:
As always: I'm more familiar with STL and I think most of the C++ developers too. I don't really understand why I should use Qt container classes outside of code that absolutely require it :)

Reply:
@FlexW But you are not using something specific to `std::vector` that does not exist in the latest `QVector`? Well, our code-base is powered by Qt, which makes perfect sense to use Qt core.

Reply:
I would like to move away from Qt containers because they make solving a problem more complex (like should I use them or a type from stl, which should I use, what performance considerations should I keep in my mind, ...)
at the moment I am fine with using both when both can be a good fit